lib/ansible/executor/module_common.py

```diff
@@ -595,39 +595,45 @@ def _get_shebang(interpreter, task_vars, templar, args=tuple()):
-
     # FUTURE: add logical equivalence for python3 in the case of py3-only modules
 
-    # check for first-class interpreter config
+    interpreter_name = os.path.basename(interpreter).strip()
+
+    # name for interpreter var
+    interpreter_config = u'ansible_%s_interpreter' % interpreter_name
+    # key for config
     interpreter_config_key = "INTERPRETER_%s" % interpreter_name.upper()
 
-    if C.config.get_configuration_definitions().get(interpreter_config_key):
-        # a config def exists for this interpreter type; consult config for the value
-        interpreter_out = C.config.get_config_value(interpreter_config_key, variables=task_vars)
-        discovered_interpreter_config = u'discovered_interpreter_%s' % interpreter_name
+    # skip detection for network os execution, use playbook supplied one if possible
+    if task_vars.get('ansible_network_os') and task_vars.get('ansible_playbook_%s' % interpreter_name):
+        interpreter_out = task_vars.get('ansible_playbook_%s' % interpreter_name)
 
-        interpreter_out = templar.template(interpreter_out.strip())
+    # a config def exists for this interpreter type; consult config for the value
+    elif C.config.get_configuration_definitions().get(interpreter_config_key):
 
-        facts_from_task_vars = task_vars.get('ansible_facts', {})
+        interpreter_from_config = C.config.get_config_value(interpreter_config_key, variables=task_vars)
+        interpreter_out = templar.template(interpreter_from_config.strip())
 
         # handle interpreter discovery if requested
         if interpreter_out in ['auto', 'auto_legacy', 'auto_silent', 'auto_legacy_silent']:
+
+            discovered_interpreter_config = u'discovered_interpreter_%s' % interpreter_name
+            facts_from_task_vars = task_vars.get('ansible_facts', {})
+
             if discovered_interpreter_config not in facts_from_task_vars:
                 # interpreter discovery is desired, but has not been run for this host
                 raise InterpreterDiscoveryRequiredError("interpreter discovery needed",
                                                         interpreter_name=interpreter_name,
                                                         discovery_mode=interpreter_out)
             else:
                 interpreter_out = facts_from_task_vars[discovered_interpreter_config]
-    else:
-        # a config def does not exist for this interpreter type; consult vars for a possible direct override
-        interpreter_config = u'ansible_%s_interpreter' % interpreter_name
 
-        if interpreter_config not in task_vars:
-            return None, interpreter
+    # lacking config (which should include vars) consult vars for a possible direct override
+    elif task_vars.get(interpreter_config):
+        interpreter_out = templar.template(task_vars.get(interpreter_config).strip())
 
-        interpreter_out = templar.template(task_vars[interpreter_config].strip())
+    else:
```

Comment:
Trying to figure out if this case might fire when we don't want it to...

Reply:
non 'python' interpreters, had an 'if interpreter_name == 'python' previouslly, but then thought that shoudl be handled at discovery side